src/jdk.jdwp.agent/share/native/libjdwp/eventFilter.c

```diff
@@ -1243,6 +1252,39 @@ matchThread(JNIEnv *env, HandlerNode *node, void *arg)
     return isSameObject(env, reqThread, goalThread);
 }
 
+// Returns true if this handler has the PlatformThreadsOnly filter enabled.
+static jboolean
+hasPlatformThreadsOnlyFilter(HandlerNode *node)
+{
+    int i;
+    Filter *filter = FILTERS_ARRAY(node);
+
+    for (i = 0; i < FILTER_COUNT(node); ++i, ++filter) {
+        switch (filter->modifier) {
+            case JDWP_REQUEST_MODIFIER(PlatformThreadsOnly):
+                return JNI_TRUE;
+            default:
+                continue;
+        }
+    }
+    return JNI_FALSE;
+}
+
+// Used to determine if no handler of the given type have the PlatformThreadsOnly filter.
+static jboolean
+matchHasNoPlatformThreadsOnlyFilter(JNIEnv *env, HandlerNode *node, void *arg)
```

Comment:
What is the reason for `arg` to be `void*`?
```suggestion
matchHasNoPlatformThreadsOnlyFilter(JNIEnv *env, HandlerNode *node, thread goalThread)
```